docker_squash/tar_image.py

```diff
@@ -0,0 +1,839 @@
+        digest = hashlib.sha256(str(to_hash).encode("utf8")).hexdigest()
+
+        return self._generate_chain_id(chain_ids, diff_ids[1:], digest)
+
+    def _write_image_metadata(self, metadata):
+        """Write image metadata and return image ID - borrowed from v2_image.py"""
+        # Create JSON from the metadata
+        # Docker adds new line at the end
+        json_metadata, image_id = self._dump_json(metadata, True)
+        image_metadata_file = os.path.join(self.new_image_dir, "%s.json" % image_id)
+
+        self._write_json_metadata(json_metadata, image_metadata_file)
+
+        return f"sha256:{image_id}"
+
+    def _dump_json(self, data, new_line=False):
+        """Dump JSON data and calculate hash - borrowed from base class"""
+        json_metadata = json.dumps(data, sort_keys=True, separators=(",", ":"))
+
+        if new_line:
+            json_metadata += "\n"
+
+        # Calculate image ID (SHA256 of the JSON metadata)
+        image_id = hashlib.sha256(json_metadata.encode()).hexdigest()
+
+        return json_metadata, image_id
+
+    def _write_json_metadata(self, metadata, metadata_file):
+        """Write JSON metadata to file - borrowed from base class"""
+        with open(metadata_file, "w") as f:
+            f.write(metadata)
+
+    def export_tar_archive(self, output_path):
+        """Export the squashed image as a tar archive"""
+        self.log.info(f"Exporting squashed image to {output_path}")
+
+        with tarfile.open(output_path, "w") as tar:
+            # Add all files from new_image_dir
+            for root, dirs, files in os.walk(self.new_image_dir):
+                for file in files:
+                    file_path = os.path.join(root, file)
+                    arcname = os.path.relpath(file_path, self.new_image_dir)
+                    tar.add(file_path, arcname=arcname)
+
+        self.log.info("Export completed successfully")
+
+    def load_squashed_image(self):
+        """Load the squashed image into Docker daemon"""
+        try:
+            from docker_squash.lib import common
+
+            docker = common.docker_client(self.log)
+
+            # Create a temporary tar file for loading
+            import tempfile
+
+            with tempfile.NamedTemporaryFile(suffix=".tar", delete=False) as temp_file:
+                temp_tar_path = temp_file.name
+
+            try:
+                # Export the squashed image to temporary tar
+                self.export_tar_archive(temp_tar_path)
+
+                # Load into Docker
+                self.log.info("Loading squashed image into Docker daemon...")
+                with open(temp_tar_path, "rb") as f:
+                    docker.load_image(f)
+
+                self.log.info("Image loaded successfully")
+
+            finally:
+                # Clean up temporary file
+                if os.path.exists(temp_tar_path):
+                    os.unlink(temp_tar_path)
+
```

Comment:
With the tarimage method surely we already havea a tar so why are we creating another one?